swagger.yaml

```diff
@@ -246,6 +246,55 @@ paths:
           description: Wrong content type (should be plain text)
         422:
           description: Missing fields
+  /projects/{projectId}/collaboration:
+    put:
+      tags:
+        - projects
+        - collaboration
+      summary: Creates an invite token for project sharing
+      security:
+        - user_id: []
+      requestBody:
+        conent:
+          application/json:
+            schema:
+              type: object
+      description: Creates a valid invite token
+      responses:
+        201:
+          description: Successfully created an invite token
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/InviteToken"
+        400:
+          description: Malformed Request
+        401:
+          description: Unauthorized
+        404:
+          description: No project found
+    post:
+      tags:
+        - projects
+        - collaboration
+      summary: Adds user to a collaboration list for a particular project
+      security:
+        - user_id: []
+      requestBody:
+        conent:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/InviteToken"
+      description: Adds users to a collaboration list if the token is not expired
+      responses:
+        201:
+          description: Successfully added user to a collaboration list
+        400:
+          description: Malformed Request
+        401:
+          description: Unauthorized
+        404:
+          description: No project found
```

Comment:
This sounds a bit weird. Some concerns:
* User shouldn't need to know `projectId` - invite token should contain such info on the backend side
* Sending token in body doesn't sounds good, it can be a parameter

Reply:
* User shouldn't need to know projectId: yes of course, we pretty much create token for that.
* I don't understand the second bullet, sending token in body doesn't sounds good, it can be a parameter. What is the difference between a parameter and a body?

Reply:
I propose to replace it with endpoint like `/share/{token}` (probably you will find out some better uri) and here token is a parameter. If you do so, to share project, user can simply generate link and give it to someone else.

Reply:
Right, i haven't noticed that a parameter is a specific name for a variable in the path. Good idea to put the token there. Already changed